mitiq/tests/test_folding_pyquil.py

```diff
@@ -1,48 +1,112 @@
 import numpy as np
 import random
 from pyquil import Program
-from pyquil.gates import X, Y, Z
-from mitiq.folding_pyquil import local_folding, unitary_folding, sampling_stretcher, left_stretcher
+from pyquil.gates import X, Y, Z, CNOT
+from pyquil.unitary_tools import program_unitary
+from mitiq.folding_pyquil import (
+    local_folding,
+    unitary_folding,
+    sampling_stretcher,
+    left_stretcher,
+)
+from copy import deepcopy
+
+
+STRETCH_VALS = [1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0]
+DEPTH = 50
+
+# NUM_SHOTS must be 1 in order to pass the next tests.
+# This is due to the fact that num_shots are not retained by
+# some methods of the Program class (bug?)
```

Comment:
Can you say a little more about what this means? What is an example method that loses the num_shots?

Reply:
This part is changed. That comment is removed.

Reply:
This part is changed. That comment is removed.